### PR TITLE
Add reg tests to NuMaker M2351 project

### PR DESCRIPTION
**Pull request by aggarg**
Description
-----------
Add reg tests to NuMaker M2351 project.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+static StackType_t xCheckTaskStack[ configMINIMAL_STACK_SIZE ] __attribute__( ( aligned( 32 ) ) );
+
+TaskParameters_t xRegTest1TaskParameters =
+{
+    .pvTaskCode      = prvRegTest1_Task,
+    .pcName          = "RegTest1",
+    .usStackDepth    = configMINIMAL_STACK_SIZE,
+    .pvParameters    = REG_TEST_1_TASK_PARAMETER,
+    .uxPriority      = tskIDLE_PRIORITY | portPRIVILEGE_BIT,
+    .puxStackBuffer  = xRegTest1TaskStack,
```
Comment (ActoryOu) [inline FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c:112]: Can this variable `xRegTest1TaskStack` be used here? It's declared in `vStartRegTests()` function scope.

**Review by sirnish — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

File: FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c (hunk 1 above)
Comment (aggarg) [reply FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c:112]: But it is static, right?

**Review by shubnil — APPROVED** [review]

File: FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c (hunk 1 above)
Comment (ActoryOu) [reply FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c:112]: It is. But is there any undefined reference happening when compiling?

Comment (ActoryOu) [reply FreeRTOS/Demo/CORTEX_MPU_M23_Nuvoton_NuMaker_PFM_M2351_IAR_GCC/RegTests/reg_tests.c:112]: Sorry, my fault. I thought `xRegTest1TaskParameters` is a different function.

**Pull request merged**